lib/rules/sort-comp.js

```diff
@@ -131,84 +131,108 @@ module.exports = {
-      }
-
-      if (method.setter) {
-        const setterIndex = methodsOrder.indexOf('setters');
-        if (setterIndex >= 0) {
-          indexes.push(setterIndex);
-        }
-      }
+      for (let groupIndex = 0; groupIndex < methodsOrder.length; groupIndex++) {
+        const currentGroup = methodsOrder[groupIndex];
 
-      if (method.typeAnnotation) {
-        const annotationIndex = methodsOrder.indexOf('type-annotations');
-        if (annotationIndex >= 0) {
-          indexes.push(annotationIndex);
-        }
-      }
-
-      if (indexes.length === 0) {
-        for (i = 0, j = methodsOrder.length; i < j; i++) {
-          isRegExp = methodsOrder[i].match(regExpRegExp);
-          if (isRegExp) {
-            matching = new RegExp(isRegExp[1], isRegExp[2]).test(method.name);
-          } else {
-            matching = methodsOrder[i] === method.name;
+        switch (currentGroup) {
+          case 'getters': {
+            if (method.getter) {
+              methodGroupIndexes.push(groupIndex);
+            }
+            break;
           }
-          if (matching) {
-            indexes.push(i);
+          case 'setters': {
+            if (method.setter) {
+              methodGroupIndexes.push(groupIndex);
+            }
+            break;
           }
-        }
-      }
-
-      if (indexes.length === 0 && method.static) {
-        const staticIndex = methodsOrder.indexOf('static-methods');
-        if (staticIndex >= 0) {
-          indexes.push(staticIndex);
-        }
-      }
-
-      if (indexes.length === 0 && method.instanceVariable) {
-        const annotationIndex = methodsOrder.indexOf('instance-variables');
-        if (annotationIndex >= 0) {
-          indexes.push(annotationIndex);
-        }
-      }
-
-      if (indexes.length === 0 && method.instanceMethod) {
-        const annotationIndex = methodsOrder.indexOf('instance-methods');
-        if (annotationIndex >= 0) {
-          indexes.push(annotationIndex);
-        }
-      }
-
-      // No matching pattern, return 'everything-else' index
-      if (indexes.length === 0) {
-        for (i = 0, j = methodsOrder.length; i < j; i++) {
-          if (methodsOrder[i] === 'everything-else') {
-            indexes.push(i);
+          case 'type-annotations': {
+            if (method.typeAnnotation) {
+              methodGroupIndexes.push(groupIndex);
+            }
+            break;
+          }
+          case 'static-methods': {
+            if (method.static) {
+              methodGroupIndexes.push(groupIndex);
+            }
+            break;
+          }
+          case 'instance-variables': {
+            if (method.instanceVariable) {
+              methodGroupIndexes.push(groupIndex);
+            }
+            break;
+          }
+          case 'instance-methods': {
+            if (method.instanceMethod) {
+              methodGroupIndexes.push(groupIndex);
+            }
+            break;
+          }
+          case 'displayName':
```

Comment:
this seems like a list of lifecycle methods that should be a shared constant array, available in multiple rules

Reply:
I think this is a sane suggestion, but...

It only works if you want to move to one of the two options above where we remove the lifecycle switch block (or `switch` completely).  

And there is code that can make use of the shared array.  (After a quick and rudimentary search I'm not seeing any other candidate code, but it may exist.)